### PR TITLE
When creating a new NHS org, set default email branding to NHS

### PR DESCRIPTION
**Pull request by CrystalPea**
Also when updating existing organisation type to an NHS type, update email branding to NHS branding if none set.

This is part of this story: https://www.pivotaltracker.com/story/show/181415981

Next step: write a migration that will update branding for existing NHS type orgs.

### CODE REVIEW

File: tests/app/organisation/test_rest.py [hunk 1]
```diff
@@ -187,6 +188,43 @@ def test_post_create_organisation(admin_request, notify_db_session, crown):
     assert len(organisation) == 1
 
 
+@pytest.mark.parametrize('org_type', ["nhs_central", "nhs_local", "nhs_gp"])
+def test_post_create_organisation_sets_default_nhs_branding_for_nhs_orgs(
+    admin_request, notify_db_session, org_type
+):
+    # we wipe email_branding table in test db between the tests, so we have to recreate this branding
+    # that is normally present on all environments and applied through migration
+    nhs_email_branding_id = current_app.config['NHS_EMAIL_BRANDING_ID']
```
Comment (benthorner) [inline tests/app/organisation/test_rest.py:197]: This looks useful but also complex - can we make it a fixture?

File: tests/app/organisation/test_rest.py [hunk 2]
```diff
@@ -187,6 +188,43 @@ def test_post_create_organisation(admin_request, notify_db_session, crown):
...
+
+    response = admin_request.post(
+        'organisation.create_organisation',
+        _data=data,
+        _expected_status=201
+    )
+
+    organisations = Organisation.query.all()
+
+    assert data['name'] == response['name']
```
Comment (benthorner) [inline tests/app/organisation/test_rest.py:219]: We don't need to test all this again as it's covered by the previous test.

File: tests/app/organisation/test_rest.py [hunk 3]
```diff
@@ -187,6 +188,43 @@ def test_post_create_organisation(admin_request, notify_db_session, crown):
     assert len(organisation) == 1
 
 
+@pytest.mark.parametrize('org_type', ["nhs_central", "nhs_local", "nhs_gp"])
```
Comment (benthorner) [inline tests/app/organisation/test_rest.py:191]: We should a test that checks that we _don't_ set this branding for other org types. I think the easiest way to do that would be to tack it on to the previous test, which is for a non-NHS org.

**Review by benthorner — APPROVED** [review]
Works locally for me ✅.

I think there's a bit more to do with the tests but it's nothing major so I'll approve now.

File: app/config.py [hunk 4]
```diff
@@ -185,6 +185,7 @@ class Config(object):
     MOU_SIGNED_ON_BEHALF_ON_BEHALF_RECEIPT_TEMPLATE_ID = '522b6657-5ca5-4368-a294-6b527703bd0b'
     NOTIFY_INTERNATIONAL_SMS_SENDER = '07984404008'
     LETTERS_VOLUME_EMAIL_TEMPLATE_ID = '11fad854-fd38-4a7c-bd17-805fb13dfc12'
+    NHS_EMAIL_BRANDING_ID = 'a7dc4e56-660b-4db7-8cff-12c37b12b5ea'
```
[the remark this answers is not in the document]
Comment (CrystalPea) [reply app/config.py:188]: I am not sure I follow. This constant is only for one particular branding (NHS branding) that we seed in our environments, and the code you are pointing at is for finding branding by name in general, which I guess would be used when we look for a branding on admin page? Why would adding the constant let us remove that dao code?

[the remark this answers is not in the document]
Comment (benthorner) [reply app/config.py:188]: That code is only used to find NHS branding, which I can see when [I search for uses of it](https://github.com/alphagov/notifications-api/search?q=dao_get_email_branding_by_name&type=). The [Platform Admin search function](https://github.com/alphagov/notifications-admin/blob/d636bd9ae06417a2bdd495aed003a74cb0dd1039/app/templates/views/email-branding/select-branding.html#L15) doesn't actually make any requests - it's just searching content on the page and isn't available with JS disabled.

[the remark this answers is not in the document]
Comment (CrystalPea) [reply app/config.py:188]: Oh wow, you are right, that's a very narrow use 😮 . I can get rid of it in a separate PR, since this one is merged now.

**Pull request merged**